tests/testthat/test-coveralls.R

```diff
@@ -43,112 +43,129 @@ ci_vars <- c(
   "WERCKER_GIT_COMMIT" = NA,
   "WERCKER_GIT_OWNER" = NA,
   "WERCKER_GIT_REPOSITORY" = NA,
-  "WERCKER_MAIN_PIPELINE_STARTED" = NA)
+  "WERCKER_MAIN_PIPELINE_STARTED" = NA
+)
 
 read_file <- function(file) paste(collapse = "\n", readLines(file))
 
-cov <- package_coverage("TestS4")
+cov <- package_coverage(test_path("TestS4"))
 
 test_that("coveralls generates a properly formatted json file", {
-
-  withr::with_envvar(c(ci_vars, "CI_NAME" = "FAKECI"),
-    with_mock(
-      `httr:::RETRY` = function(...) list(...),
-      `httr::content` = identity,
-      `httr::upload_file` = function(file) readChar(file, file.info(file)$size),
-
-      res <- coveralls(coverage = cov),
-      json <- jsonlite::fromJSON(res$body$json_file),
-
-      expect_equal(nrow(json$source_files), 1),
-      expect_equal(json$service_name, "fakeci"),
-      expect_match(json$source_files$name, rex::rex("R", one_of("/", "\\"), "TestS4.R")),
-      expect_equal(json$source_files$source, read_file("TestS4/R/TestS4.R")),
-      expect_equal(json$source_files$source_digest, '1233f2eca5d84704101cb9d9b928f2e9'),
-      expect_equal(json$source_files$coverage[[1]],
-        c(NA, NA, NA, NA, NA, NA, 5, 2, NA, 3, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA,
-          NA, NA, NA, NA, 1, NA, NA, NA, NA, NA, 1, NA, NA, NA, NA, NA, 1, NA))
+  withr::with_envvar(
+    c(ci_vars, "CI_NAME" = "FAKECI"),
+    with_mocked_bindings(
+      RETRY = function(...) list(...),
+      content = identity,
+      upload_file = function(file) readChar(file, file.info(file)$size),
+      code = {
+        res <- coveralls(coverage = cov)
+        json <- jsonlite::fromJSON(res$body$json_file)
+
+        expect_equal(nrow(json$source_files), 1)
+        expect_equal(json$service_name, "fakeci")
+        expect_match(json$source_files$name, rex::rex("R", one_of("/", "\\"), "TestS4.R"))
+        expect_equal(json$source_files$source, read_file("TestS4/R/TestS4.R"))
+        expect_equal(json$source_files$source_digest, "1233f2eca5d84704101cb9d9b928f2e9")
+        expect_equal(
+          json$source_files$coverage[[1]],
+          c(
+            NA, NA, NA, NA, NA, NA, 5, 2, NA, 3, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA,
+            NA, NA, NA, NA, 1, NA, NA, NA, NA, NA, 1, NA, NA, NA, NA, NA, 1, NA
+          )
+        )
+      }
     )
   )
 })
```

Comment:
An example of replacing `with_mock()` with `with_mocked_bindings()`.

In short, we cannot have namespaced functions. And we give it a series of functions to mock. And then provide which code to run with the mocking. Very similar to `withr::with_*()` syntax.